rust_snuba/src/strategies/commit_log.rs

```diff
@@ -81,6 +92,105 @@ impl TryFrom<Commit> for KafkaPayload {
     }
 }
 
+struct ProduceMessage {
+    producer: Arc<dyn Producer<KafkaPayload>>,
+    topic: Arc<TopicOrPartition>,
+    skip_produce: bool,
+}
+
+impl ProduceMessage {
+    #[allow(dead_code)]
+    pub fn new(
+        producer: impl Producer<KafkaPayload> + 'static,
+        topic: TopicOrPartition,
+        skip_produce: bool,
+    ) -> Self {
+        ProduceMessage {
+            producer: Arc::new(producer),
+            topic: Arc::new(topic),
+            skip_produce,
+        }
+    }
+}
+
+impl TaskRunner<KafkaPayload, KafkaPayload> for ProduceMessage {
+    fn get_task(&self, message: Message<KafkaPayload>) -> RunTaskFunc<KafkaPayload> {
+        let producer = self.producer.clone();
+        let topic = self.topic.clone();
+        let skip_produce = self.skip_produce;
+
+        Box::pin(async move {
```

Comment:
What do you achieve with this? (pinning the function here)

Reply:
We cannot spawn the future without pin, and it's part of the RunTaskInThreads interface.

https://rust-lang.github.io/async-book/04_pinning/01_chapter.html is probably a better explanation than I am able to give.